File: Python/isolate_duplicates.py
test = "aaaabbcdefffffffg"
# ["aa[aa]bbcdeff[fffff]g", 2]

def isolate(duplicates):
    words = []
    initial = []
    hold = ""
    counter = 0
    # For loop to get the uniques of the letters
    for x in duplicates:
        if x not in initial:
            initial.append(x)
    # Loop to get similar characters in a single array
    for x in initial:
        for char in duplicates:
            if char == x:
                hold += x
        words.append(hold)
        hold = ""
    # Loop to arrange them in patterns
    for x in words:
        if len(x) > 2:
            hold += f"{x[:2]}[{x[2:]}]"
            counter += 1
        else:
            hold += x

    return [hold, counter]


print(isolate("aaasssssssddddffgkkkkkuueyyryrhhsgfhjjsnnnnnnjjfmgjghjdjjjjdhyteghsjckjdheyebnfjhhgnjdfyfggbejjwtyjfjcbdgdhdjfjeeebddhdychdjdjmkj"))

File: Javascript/isolate_duplicates.js
test = "aaaabbcdefffffffg"
//  ["aa[aa]bbcdeff[fffff]g", 2]

function isolate(){
    let words = [];
    let initial = [];
    let hold = "";
    let counter = 0;

    for (i in test){
        if (!initial.includes(test[i])){
                initial.push(test[i]);
        }
    }

    for (x in initial) {
        for (char in test) {
            if (test[char] == initial[x]){
                hold += initial[x];
            }
        }
        words.push(hold);
        hold = "";
    }

    for (x in words){
        if (words[x].length > 2){
            hold += words[x].slice(0,2).concat('[', words[x].slice(2), ']');
            counter += 1;
        } else{
            hold += words[x];
        }
    }

    return [hold, counter];


}

console.log(isolate("aaasssssssddddffgkkkkkuueyyryrhhsgfhjjsnnnnnnjjfmgjghjdjjjjdhyteghsjckjdheyebnfjhhgnjdfyfggbejjwtyjfjcbdgdhdjfjeeebddhdychdjdjmkj"))